autogpt_platform/frontend/src/components/admin/spending/add-credit-button.tsx

```diff
@@ -0,0 +1,146 @@
+}: {
+  userId: string;
+  userEmail: string;
+  currentBalance: number;
+  defaultAmount?: number;
+  defaultComments?: string;
+}) {
+  const router = useRouter();
+  const [isAddCreditDialogOpen, setIsAddCreditDialogOpen] = useState(false);
+  const [dollarAmount, setDollarAmount] = useState(
+    defaultAmount ? Math.abs(defaultAmount / 100).toFixed(2) : "1.00",
+  );
+
+  // Calculate credits from dollar amount
+  const calculateCredits = (dollars: string) => {
+    const amount = Number.parseFloat(dollars) || 0;
+    return Math.round(amount * 100);
+  };
+
+  const handleApproveSubmit = async (formData: FormData) => {
+    setIsAddCreditDialogOpen(false);
+    try {
+      await addDollars(formData);
+      router.refresh(); // Refresh the current route
+    } catch (error) {
+      console.error("Error adding dollars:", error);
+    }
+  };
+
+  return (
+    <>
+      <Button
+        size="sm"
+        variant="default"
+        onClick={(e) => {
+          e.stopPropagation();
+          setIsAddCreditDialogOpen(true);
+        }}
+      >
+        Add Dollars
+      </Button>
+
+      {/* Add $$$ Dialog */}
+      <Dialog
+        open={isAddCreditDialogOpen}
+        onOpenChange={setIsAddCreditDialogOpen}
+      >
+        <DialogContent>
+          <DialogHeader>
+            <DialogTitle>Add Dollars</DialogTitle>
+            <DialogDescription className="pt-2">
+              <div className="mb-2">
+                <span className="font-medium">User:</span> {userEmail}
+              </div>
+              <div>
+                <span className="font-medium">Current balance:</span> $
+                {(currentBalance / 100).toFixed(2)}
+              </div>
+            </DialogDescription>
+          </DialogHeader>
+
+          <form action={handleApproveSubmit}>
+            <input type="hidden" name="id" value={userId} />
+            <input
+              type="hidden"
+              name="amount"
+              value={calculateCredits(dollarAmount)}
+            />
+
+            <div className="grid gap-4 py-4">
+              <div className="grid gap-2">
+                <Label htmlFor="dollarAmount">Amount (in dollars)</Label>
+                <div className="flex">
+                  <div className="flex items-center justify-center rounded-l-md border border-r-0 bg-gray-50 px-3 text-gray-500">
+                    $
+                  </div>
+                  <Input
+                    id="dollarAmount"
+                    type="number"
+                    step="0.01"
+                    min="0"
+                    className="rounded-l-none"
+                    value={dollarAmount}
+                    onChange={(e) => setDollarAmount(e.target.value)}
+                    placeholder="0.00"
+                  />
+                </div>
+                <p className="text-xs text-gray-500">
+                  This will add {calculateCredits(dollarAmount)} credits (1
+                  credit = 1¢)
```

Comment:
There is `formatCredits` in useCredits

Reply:
Credits is no longer a thing, we don't show number of credits anywhere, and there is only dollar amount.